tests/devices/i10/test_diagnostics.py

```diff
@@ -0,0 +1,66 @@
+from collections import defaultdict
+from unittest.mock import Mock
+
+import pytest
+from bluesky.plans import count
+from bluesky.run_engine import RunEngine
+from ophyd_async.core import (
+    DeviceCollector,
+)
+from ophyd_async.testing import (
+    assert_emitted,
+    callback_on_mock_put,
+    set_mock_value,
+)
+
+from dodal.devices.i10.diagnostics import I10CentroidDetector
+
+
+@pytest.fixture
+async def mock_centroid_detector(
+    prefix: str = "BLXX-EA-DET-007:",
+) -> I10CentroidDetector:
+    async with DeviceCollector(mock=True):
+        mock_centroid_detector = I10CentroidDetector(prefix)
+    assert mock_centroid_detector.name == "mock_centroid_detector"
+    return mock_centroid_detector
+
+
+async def test_I10CentroidDetector_with_blueSky(
```

Comment:
Nit: My preference is to name tests very verbosely and explicitly so it's clear what you're testing. For example here I would name it something like `test_when_I10CentroidDetector_is_counted_in_plan_then_emitted_document_contains_expected_values`. If you have vague names I think it leads to the temptation to make one big test that tests loads of different stuff, that test then becomes long to run and hard to debug